lib/lambda_ethereum_consensus/fork_choice/store.ex

```diff
@@ -91,14 +96,17 @@ defmodule LambdaEthereumConsensus.ForkChoice.Store do
            # process block attestations
            {:ok, new_state} <-
              signed_block.message.body.attestations
-             |> Enum.reduce_while({:ok, new_state}, fn
-               att, {:ok, st} -> {:cont, Handlers.on_attestation(st, att, true)}
-               _, {:error, _} = err -> {:halt, err}
-             end) do
+             |> apply_handler(new_state, &Handlers.on_attestation(&1, &2, true)),
+           {:ok, new_state} <-
+             signed_block.message.body.attester_slashings
+             |> apply_handler(new_state, &Handlers.on_attester_slashing/2) do
         new_state
       else
         _ ->
-          Logger.error("Failed to add block #{signed_block.message.slot} to the store.")
+          Logger.error(
```

Comment:
should we add the error reason to the log?

Reply:
Done